ppl/src/main/java/org/opensearch/sql/ppl/parser/AstBuilder.java

```diff
@@ -309,7 +309,7 @@ protected UnresolvedPlan aggregateResult(UnresolvedPlan aggregate, UnresolvedPla
 
   @Override
   public UnresolvedPlan visitKmeansCommand(KmeansCommandContext ctx) {
-    return new Kmeans(ArgumentFactory.getArgumentList(ctx));
+    return new Kmeans(ArgumentFactory.getArgumentMap(ctx));
```

Comment:
Could you try to avoid if..else in getArgumentMap() by add label to antlr and using visitor to access each arguments? https://github.com/antlr/antlr4/blob/master/doc/parser-rules.md#alternative-labels

Reply:
Tweaked the code by adding label and using visitor